Can't retrieve commits in native mode

While trying to retrieve a commit, I got the following error:

2021-05-21T07:38:50.557926+00:00 app[web.1]: 2021-05-21 07:38:50,557 DEBUG [me.xun.ArtifactProcessor] (executor-thread-1) Processing with scanner: me.xunitreporter.spi.SurefireArtifactScanner@346ae064
2021-05-21T07:38:50.706201+00:00 app[web.1]: 2021-05-21 07:38:50,705 ERROR [io.qua.githubapp] (executor-thread-1) Error handling delivery 9566c5d0-ba07-11eb-995c-d8328330662e
2021-05-21T07:38:50.706202+00:00 app[web.1]: › Repository: xunit-reporter/test-xunit
2021-05-21T07:38:50.706203+00:00 app[web.1]: › Event:      workflow_run.completed
2021-05-21T07:38:50.706207+00:00 app[web.1]: Exception: java.util.concurrent.CompletionException: org.kohsuke.github.HttpException: Server returned HTTP response code: 200, message: 'null' for URL: https://api.github.com/repos/xunit-reporter/test-xunit/commits/bfbdf1729cea6d900f4e8006ce379a2ff60ff5c1
2021-05-21T07:38:50.706207+00:00 app[web.1]: at io.quarkus.arc.impl.EventImpl.handleExceptions(EventImpl.java:192)
2021-05-21T07:38:50.706210+00:00 app[web.1]: at io.quarkus.arc.impl.EventImpl.access$000(EventImpl.java:47)
2021-05-21T07:38:50.706210+00:00 app[web.1]: at io.quarkus.arc.impl.EventImpl$1.get(EventImpl.java:99)
2021-05-21T07:38:50.706213+00:00 app[web.1]: at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
2021-05-21T07:38:50.706214+00:00 app[web.1]: at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
2021-05-21T07:38:50.706214+00:00 app[web.1]: at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
2021-05-21T07:38:50.706214+00:00 app[web.1]: at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
2021-05-21T07:38:50.706214+00:00 app[web.1]: at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
2021-05-21T07:38:50.706214+00:00 app[web.1]: at java.lang.Thread.run(Thread.java:829)
2021-05-21T07:38:50.706215+00:00 app[web.1]: at org.jboss.threads.JBossThread.run(JBossThread.java:501)
2021-05-21T07:38:50.706215+00:00 app[web.1]: at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:553)
2021-05-21T07:38:50.706215+00:00 app[web.1]: at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:192)
2021-05-21T07:38:50.706221+00:00 app[web.1]: Caused by: org.kohsuke.github.HttpException: Server returned HTTP response code: 200, message: 'null' for URL: https://api.github.com/repos/xunit-reporter/test-xunit/commits/bfbdf1729cea6d900f4e8006ce379a2ff60ff5c1
2021-05-21T07:38:50.706221+00:00 app[web.1]: at org.kohsuke.github.GitHubClient.interpretApiError(GitHubClient.java:490)
2021-05-21T07:38:50.706222+00:00 app[web.1]: at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:409)
2021-05-21T07:38:50.706222+00:00 app[web.1]: at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:357)
2021-05-21T07:38:50.706222+00:00 app[web.1]: at org.kohsuke.github.Requester.fetch(Requester.java:76)
2021-05-21T07:38:50.706222+00:00 app[web.1]: at org.kohsuke.github.GHRepository.getCommit(GHRepository.java:1879)
2021-05-21T07:38:50.706223+00:00 app[web.1]: at me.xunitreporter.XUnitBot.onWorkflow(XUnitBot.java:56)
2021-05-21T07:38:50.706225+00:00 app[web.1]: at me.xunitreporter.XUnitBot_Multiplexer.onWorkflow_5faf52dd037aecb35cbd62097faa01b3382161b8(XUnitBot_Multiplexer.zig:48)
2021-05-21T07:38:50.706226+00:00 app[web.1]: at me.xunitreporter.XUnitBot_Multiplexer_Observer_onWorkflow_5faf52dd037aecb35cbd62097faa01b3382161b8_de3cdc797744b18c5a527cdfb99ea3e1dd778a7f.notify(XUnitBot_Multiplexer_Observer_onWorkflow_5faf52dd037aecb35cbd62097faa01b3382161b8_de3cdc797744b18c5a527cdfb99ea3e1dd778a7f.zig:221)
2021-05-21T07:38:50.706226+00:00 app[web.1]: at io.quarkus.arc.impl.EventImpl$Notifier.notifyObservers(EventImpl.java:283)
2021-05-21T07:38:50.706226+00:00 app[web.1]: at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:268)
2021-05-21T07:38:50.706226+00:00 app[web.1]: at io.quarkus.arc.impl.EventImpl$1.get(EventImpl.java:98)
2021-05-21T07:38:50.706227+00:00 app[web.1]: ... 9 more
2021-05-21T07:38:50.706227+00:00 app[web.1]: Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `org.kohsuke.github.GHCommit$ShortInfo$Tree` (no Creators, like default constructor, exist): cannot deserialize from Object value (no delegate- or property-based Creator)
2021-05-21T07:38:50.706229+00:00 app[web.1]: at [Source: (String)"{"sha":"bfbdf1729cea6d900f4e8006ce379a2ff60ff5c1","node_id":"MDY6Q29tbWl0MzUyMTMwNzI0OmJmYmRmMTcyOWNlYTZkOTAwZjRlODAwNmNlMzc5YTJmZjYwZmY1YzE=","commit":{"author":{"name":"Jeff MAURY","email":"jmaury@redhat.com","date":"2021-05-21T07:37:53Z"},"committer":{"name":"GitHub","email":"noreply@github.com","date":"2021-05-21T07:37:53Z"},"message":"Update dummy (17)","tree":{"sha":"1f69caee34a51ecce29735300c77387356498478","url":"https://api.github.com/repos/xunit-reporter/test-xunit/git/trees/1f69caee34"[truncated 3982 chars]; line: 1, column: 370] (through reference chain: org.kohsuke.github.GHCommit["commit"]->org.kohsuke.github.GHCommit$ShortInfo["tree"])
2021-05-21T07:38:50.706230+00:00 app[web.1]: at com.fasterxml.jackson.databind.DeserializationContext.reportBadDefinition(DeserializationContext.java:1764)
2021-05-21T07:38:50.706230+00:00 app[web.1]: at com.fasterxml.jackson.databind.DatabindContext.reportBadDefinition(DatabindContext.java:400)
2021-05-21T07:38:50.706230+00:00 app[web.1]: at com.fasterxml.jackson.databind.DeserializationContext.handleMissingInstantiator(DeserializationContext.java:1209)
2021-05-21T07:38:50.706230+00:00 app[web.1]: at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1400)
2021-05-21T07:38:50.706234+00:00 app[web.1]: at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:362)
2021-05-21T07:38:50.706234+00:00 app[web.1]: at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:195)
2021-05-21T07:38:50.706235+00:00 app[web.1]: at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:138)
2021-05-21T07:38:50.706235+00:00 app[web.1]: at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:324)
2021-05-21T07:38:50.706235+00:00 app[web.1]: at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:187)
2021-05-21T07:38:50.706235+00:00 app[web.1]: at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:138)
2021-05-21T07:38:50.706235+00:00 app[web.1]: at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:324)
2021-05-21T07:38:50.706236+00:00 app[web.1]: at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:187)
2021-05-21T07:38:50.706236+00:00 app[web.1]: at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:322)
2021-05-21T07:38:50.706236+00:00 app[web.1]: at com.fasterxml.jackson.databind.ObjectReader._bindAndClose(ObjectReader.java:2033)
2021-05-21T07:38:50.706236+00:00 app[web.1]: at com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:1491)
2021-05-21T07:38:50.706236+00:00 app[web.1]: at org.kohsuke.github.GitHubResponse.parseBody(GitHubResponse.java:97)
2021-05-21T07:38:50.706236+00:00 app[web.1]: at org.kohsuke.github.Requester.lambda$fetch$1(Requester.java:76)
2021-05-21T07:38:50.706237+00:00 app[web.1]: at org.kohsuke.github.GitHubClient.createResponse(GitHubClient.java:451)
2021-05-21T07:38:50.706237+00:00 app[web.1]: at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:401)
2021-05-21T07:38:50.706237+00:00 app[web.1]: ... 18 more
2021-05-21T07:38:50.706237+00:00 app[web.1]:

--- a/deployment/src/main/java/io/quarkiverse/githubapi/deployment/GitHubApiDotNames.java
+++ b/deployment/src/main/java/io/quarkiverse/githubapi/deployment/GitHubApiDotNames.java
@@ -115,7 +115,7 @@ final class GitHubApiDotNames {
     private static final DotName GH_COMMIT_PARENT = DotName.createSimple(GHCommit.Parent.class.getName());
     private static final DotName GH_COMMIT_SHORT_INFO = DotName.createSimple(GHCommit.ShortInfo.class.getName());
     private static final DotName GH_COMMIT_STATS = DotName.createSimple(GHCommit.Stats.class.getName());
-    private static final DotName GH_COMMIT_TREE = DotName.createSimple(GHCommit.class.getName() + "$Tree");
+    private static final DotName GH_COMMIT_TREE = DotName.createSimple(GHCommit.ShortInfo.class.getName() + "$Tree");
     private static final DotName GH_COMMIT_USER = DotName.createSimple(GHCommit.class.getName() + "$User");
     private static final DotName GH_COMMIT_POINTER = DotName.createSimple(GHCommitPointer.class.getName());
     private static final DotName GH_COMPARE = DotName.createSimple(GHCompare.class.getName());

